.github/workflows/publish.yml

```diff
@@ -13,15 +13,16 @@ jobs:
       - uses: ruby/setup-ruby@v1
         with:
           bundler-cache: true
+          working-directory: legacy
       - uses: actions/cache@v1
         with:
-          path: vendor/bundle
+          path: legacy/vendor/bundle
           key: ${{ runner.os }}-gems-${{ hashFiles('**/Gemfile.lock') }}
           restore-keys: |
             ${{ runner.os }}-gems-
       - run: gem install bundler
-      - run: bundle install
-      - run: bundle exec jekyll build
+      - run: cd legacy && bundle install
+      - run: cd legacy && bundle exec jekyll build
```

Comment:
If you wrap the yarn and jekyll builds in a build.sh then the build command is the same for Github, local, and Netlify. In the netlify toml you can define a custom build command and set the ruby and node versions.